application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubReference.java

```diff
@@ -0,0 +1,168 @@
+package org.togetherjava.tjbot.commands.github;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import org.apache.commons.collections4.ListUtils;
+import org.kohsuke.github.*;
+import org.togetherjava.tjbot.annotations.MethodsReturnNonnullByDefault;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.awt.*;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.List;
+
+/**
+ * GitHub Referencing feature. If someone sends #id of an issue (e.g. #207) the bot replies with an
+ * embed that contains info on the issue/pr
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class GitHubReference extends MessageReceiverAdapter {
+    /**
+     * The pattern used to determine whether a message is referencing an issue
+     */
+    protected static final Pattern ISSUE_REFERENCE_PATTERN = Pattern.compile("#(\\d+)");
+
+    private final Config config;
+
+    /**
+     * The repositories that are searched when looking for an issue
+     */
+    private List<GHRepository> repositories;
+
+    public GitHubReference(Config config) {
+        super(Pattern.compile(config.getGitHubReferenceChannelPattern()));
+
+        this.config = config;
+
+        try {
+            acquireRepositories();
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
```

Comment:
just move the try-catch stuff into the helper method, keeps the constructor clean

Reply:
done